src/data/chats/mod.rs

```diff
@@ -225,4 +261,113 @@ impl Chats {
+        
+        self.mofa_servers.insert(server_id.clone(), MofaServer {
+            address: address.clone(),
+            client,
+            connection_status: MofaServerConnectionStatus::Connecting,
+        });
+
+        self.test_mofa_server_and_fetch_agents(&address);
+        server_id
+    }
+
+    /// Removes a MoFa server from the list of available servers.
+    pub fn remove_mofa_server(&mut self, address: &str) {
+        self.mofa_servers.remove(&MofaServerId(address.to_string()));
+        self.available_agents.retain(|_, agent| agent.server_id.0 != address);
+    }
+
+    /// Retrieves the corresponding MofaClient for an agent
+    pub fn get_client_for_agent(&self, agent_id: &AgentId) -> Option<Rc<MofaClient>> {
+        self.available_agents.get(agent_id)
+            .and_then(|agent| self.mofa_servers.get(&agent.server_id))
+            .map(|server| server.client.clone())
+    }
+
+    /// Helper method for components that need a sorted vector of agents
+    pub fn get_agents_list(&self) -> Vec<MofaAgent> {
+        let mut agents: Vec<_> = self.available_agents.values().cloned().collect();
+        agents.sort_by(|a, b| a.name.cmp(&b.name));
+        agents
+    }
+
+    /// Tests the connection to a MoFa server by requesting /v1/models
+    /// The connection status is updated at the App level based on the actions dispatched
+    /// 
+    /// We perform fetching (and therefore testing) as follows:
+    /// 1. We create a channel used to communicate with a background thread that will do the http request
+    /// 2. We start that background thread and instruct it to fetch the agents from the server
+    /// 3. The thread sends a message back to the channel created here with the server response
+    /// 4. We forward that result back to the UI thread using Cx::post_action
+    /// 5. The action is handled at the App level, which tells the store to update the connection status and agent list
+    pub fn test_mofa_server_and_fetch_agents(&mut self, address: &String) {
+        self.mofa_servers.get_mut(&MofaServerId(address.to_string())).unwrap().connection_status = MofaServerConnectionStatus::Connecting;
+        let (tx, rx) = mpsc::channel();
+        if let Some(server) = self.mofa_servers.get(&MofaServerId(address.to_string())) {
+            server.client.fetch_agents(tx.clone());
+        }
+
+        std::thread::spawn(move || match rx.recv() {
+            Ok(MofaServerResponse::Connected(server_address, agents)) => {
+                Cx::post_action(MoFaTestServerAction::Success(server_address, agents));
+            }
+            Ok(MofaServerResponse::Unavailable(server_address)) => {
+                Cx::post_action(MoFaTestServerAction::Failure(Some(server_address)));
+            }
+            Err(e) => {
+                error!("Error receiving response from MoFa backend: {:?}", e);
+                Cx::post_action(MoFaTestServerAction::Failure(None));
+            }
+        });
+    }
```

Comment:
Combined server testing and agent fetching into a single operation to simplify async flow.

Previously, we had separate operations for testing server connectivity and fetching agents (which would both just do the same thing, fetching `/v1/models`). 
This created complexity in managing multiple async operations without proper abstractions for 
state management. By combining these into a single operation, we:

1. Reduce complexity in handling async state transitions
2. Maintain consistency between server status and available agents
3. Simplify the UI feedback loop
4. Make the transition to real server responses easier when they become available